### DISCUSSION TITLE
What's the correct FusionCache configuration for this scenario? (No expiration + distributed invalidation)

### DISCUSSION THREAD

**Discussion by FrancescoBonizzi** [ZiggyCreatures/FusionCache#125] [Q&A] [answered]
Hi! Imagine this scenario:
- You have _n_ web server
- Each web server caches a key value pair collection called `FeatureFlags`
- Those values never changes until a person changes it manually from _a secret back office_, for example he could change the value of the feature flag `IsFeatureXyEnabled`
- The distributed cache is configured and active

Ideally I'd like that:
- On each web server **the local layer** caches the value **forever** (`options => options.SetDuration(TimeSpan.MaxValue)`) which currently throws), since the value can be changed only manually and I'd like the value to be refreshed only when I'm sure it is changed.
- When the man changes and saves the feature flag from the _secret backoffice_, the distributed cache invalidates that value, and all the web server get notified and refresh its value.

How do you configuration this scenario with `FusionCache`?

**Comment by jodydonetti** [comment, marked as answer]
Hi @FrancescoBonizzi and thanks for using FusionCache!

Yes, you can achieve what you are describing 🎉 
But no, not with the distributed cache.

You see, a cache layer (both memory and distributed) is "passive" whereas you need something "active", meaning that each node gets to be notified of the change, right?
Well you are in luck: that is exactly the job of another component, the [backplane](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/Backplane.md).
Basically you just need to enable the backplane and... that's it: when you'll change a piece of data, a notification will be sent to all the connected nodes.
You can even enable/disable backplane notifications per-call, with a very granular control.

Currently the only current implementation available is the one for Redis, but any implementation of the `IFusionCacheBackplane` interface would work.

Please take a look at the docs I just linked, and tell me if you need anything else.

**Reply by FrancescoBonizzi** [reply]
Thanks a lot, for the very useful library and for your answer. Good night XD

**Reply by jodydonetti** [reply]
No problem, hope this helped!

Let me know if you have any trouble using it, and if you'll be able to achieve what you are trying to do.

ps: just to be clear, when I said _"not with the distributed cache"_, I meant that to have the notifications you need to use the backplane, and not that you cannot use the distributed cache too (in conjunction with the backplane): you can use them both together. Just wanted to clarify this.

**Reply by FrancescoBonizzi** [reply]
I read the link you posted and it's absolutely clear. I understood, tomorrow I will implement it with my solution, I'm going to start with Redis on Azure with the distributed cache too since you suggest it's less fragile.
